source/postprocess/particle_density_histogram.cc

```diff
@@ -0,0 +1,268 @@
+      statistics.add_value ("Maximal density gradient score: ", global_max_score);
+
+      std::ostringstream output;
+      output << global_min_score << "/ " <<average_score << "/ " << global_max_score;
+
+      return std::pair<std::string, std::string> ("Particle density gradient score min/avg/max:",
+                                                  output.str());
+    }
+
+
+
+    template <int dim>
+    std::list<std::string>
+    ParticleDensityStatistics<dim>::required_other_postprocessors() const
+    {
+      return {"particles"};
+    }
+
+    
+    template <int dim>
+    void ParticleDensityStatistics<dim>::sortParticles(const typename Triangulation<dim>::active_cell_iterator &cell,Table<dim,unsigned int> &buckets,double bucket_width)
+    {
+      for (unsigned int particle_manager_index = 0; particle_manager_index < this->n_particle_managers(); ++particle_manager_index)
+      {
+        //sort only the particles within the cell 
+        auto first_particle_in_cell = this->get_particle_manager(particle_manager_index).get_particle_handler().particles_in_cell(cell).begin();
+        auto last_particle_in_cell = this->get_particle_manager(particle_manager_index).get_particle_handler().particles_in_cell(cell).end();
+        
+        for(auto particle_iterator=first_particle_in_cell; particle_iterator != last_particle_in_cell; std::advance(particle_iterator,1))
+        {
+          double particle_x = particle_iterator->get_reference_location()[0];
+          double particle_y = particle_iterator->get_reference_location()[1];
+
+          double x_ratio = (particle_x) / (bucket_width);
+          double y_ratio = (particle_y) / (bucket_width);
+
+          unsigned int x_index = static_cast<unsigned int>(std::floor(x_ratio));
+          unsigned int y_index = static_cast<unsigned int>(std::floor(y_ratio));
+
+          /*
+          if a particle is exactly on the boundary of two cells it's 
+          reference location will equal 1 and if this is the case
+          the x/y/z_index will be outside of the range of the table without
+          these checks.
+          */
+          if (x_index == (dim-1)){
+            x_index = dim-1;
+          }
+          if (y_index == (dim-1)){
+            y_index = dim-1;
+          }
+
+          TableIndices<dim> entry_index;
+          entry_index[0] = x_index;
+          entry_index[1] = y_index;
+          if (dim == 3){
+            const double particle_z = particle_iterator->get_reference_location()[2];
+            double z_ratio = (particle_z) / (bucket_width);
+            unsigned int z_index = static_cast<unsigned int>(std::floor(z_ratio));
+            if (z_index == (dim-1)){
+              z_index = dim-1;
+            }
+            entry_index[2] = z_index;
+          }
+
+          unsigned int particles_in_bucket = buckets(entry_index);
+          particles_in_bucket++;
+          buckets(entry_index) = particles_in_bucket;    
```

Comment:
would this work:
```suggestion
          ++buckets(entry_index);    
```
or if not, at least this:
```suggestion
          buckets(entry_index) += 1;    
```